### ISSUE TITLE
Differences with VQ-GAN?

### ISSUE THREAD

**Issue by BrandonHanx** [microsoft/PeCo#1]
Hi, thanks for your work.

I was just wondering what's the exact difference between the famous VQ-GAN and your proposed perceptual codebook learning? Just without a GAN loss?

As you mentioned in the paper "it can be easily seen that with the adversarial loss, the image looks more photo-realistic, presenting finer details," and your w/o GAN-loss training didn't achieve much better performance compared to w/ GAN-loss training.
So why we need w/o GAN-loss training?

Please point me out if I were wrong.

**Comment by jianminbao** [comment]
Hi, thanks for your interest in our work.

Our work PeCo is fundamentally different from VQ-GAN in 3 aspects.

1. Different motivations, the discrete tokens (e.g., BPE) in the NLP field are naturally highly semantic and very helpful for pre-training. But how to discretize images that can benefit vision pre-training is still an open question. Our proposed PeCo aims to design a new codebook for BERT pre-training of Vision Transformers and demonstrates that a better vision tokenizer is necessary. In contrast, VQ-GAN aims to improve the quality of image generation. 

2. VQ-GAN applies the VGG model to calculate the perceptual loss, while VGG is a supervised model, which is not suitable for self-supervised learning. We study this issue and find that self-supervised models are well-suited for learning a new codebook.

3. We empirically show that the reconstructed image quality is irrelevant for pre-training. The VQ-GAN can really produce high-quality images while this isn't beneficial for codebook of pre-training. But we may suffer from two issues: discriminator can introduce extra computation cost and instability of training process.

**Comment by BrandonHanx** [comment]
Hi, thanks for your reply.

1. I agree that the motivations are different. But I guess the implementations are very similar...
2. Totally agree. But I am afraid you need to cite VQGAN in the main paper at least ...
3. I agree `discriminator can introduce extra computation cost and instability of training process`. But as you implemented in Table 10, this issue seems like can be addressed easily... Although VQ-GAN is not beneficial for the codebook, the quality of reconstructed images is also important if we want to extend this framework into some generative tasks. Right?
